handlers.js

```diff
@@ -94,17 +94,167 @@
+                // check if file exists...
+                this.fs.getByPath(this.request.path, this.onPutEntry.bind(this), true)
             }
-
-            // if upload enabled in options...
-            // check if file exists...
-            this.fs.getByPath(this.request.path, this.onPutEntry.bind(this), true)
+            function putCheck() {
+                if (! this.app.opts.optUpload) {
+                    this.responseLength = 0
+                    this.writeHeaders(400)
+                    this.finish()
+                    return
+                } else {
+                    putMain.bind(this)()
+                }
+            }
+            
+            if (this.app.opts.optScanForHtaccess) {
+                var fullrequestpath = this.request.origpath
+                    var finapath = fullrequestpath
+                    var finpath = finapath.split('/').pop();
+                    var finalpath = fullrequestpath.substring(0, fullrequestpath.length - finpath.length);
+                if (this.request.path == '') {
+                    var finalpath = '/'
+                }
+                var htaccesspath = finalpath+'wsc.htaccess'
+                //console.log(htaccesspath)
+                this.fs.getByPath(htaccesspath, (file) => {
+                if (! file.error) {
+                    file.file( function(filee) {
+                        var reader = new FileReader();
+                        reader.onload = function(e){
+                            var dataa = e.target.result
+                            if(true) {
+                                try {
+                                    var origdata = JSON.parse(dataa)
+                                } catch(e) {
+                                    this.responseLength = 0
+                                    this.writeHeaders(500)
+                                    this.finish()
+                                    return
+                                }
+                            }
+                            var filerequest = this.request.origpath
+                            var filerequested = filerequest.split('/').pop();
+                            var filefound = false
+                            for (var i=0; i<origdata.length; i++) {
+                                if ((origdata[i].type == 'allow put' && origdata[i].request_path == filerequested) ||
+                                    (origdata[i].type == 'allow put' && origdata[i].request_path == 'all files') ||
+                                    (origdata[i].type == 'deny put' && origdata[i].request_path == filerequested) ||
+                                    (origdata[i].type == 'deny put' && origdata[i].request_path == 'all files')) {
+                                    var data = origdata[i]
+                                    var filefound = true
+                                    break
+                                }
+                            }
```

Comment:
it looks like there's a lot of code duplication between delete and put access checks. It could probably be simplified if you wanted to (I don't care)

Reply:
I thought about doing that, but It would take a lot of works. It works perfectly at the moment so I dont plan on changing it.